pyglotaran_extras/plotting/plot_traces.py

```diff
@@ -13,14 +44,22 @@ def get_shifted_traces(res, center_λ=None):
     times = traces.coords["time"]
     if center_λ is None:  # center wavelength (λ in nm)
         center_λ = min(res.dims["spectral"], round(res.dims["spectral"] / 2))
-    if "center_dispersion_1" in res:
-        center_dispersion = res.center_dispersion_1  # TODO: clarify against pyglotaran API why _1?
+
+    if "irf_center_location" in res:
+        center_dispersion = res.center_dispersion
+        irf_loc = center_dispersion.sel(spectral=center_λ, method="nearest").item()
+    elif "center_dispersion_1" in res:
+        # legacy compatibility pyglotaran<0.5.0
+        center_dispersion = res.center_dispersion_1
         irf_loc = center_dispersion.sel(spectral=center_λ, method="nearest").item()
     elif "irf_center" in res:
         irf_loc = res.irf_center
     else:
         irf_loc = min(times)
 
+    if hasattr(irf_loc, "shape"):
+        irf_loc = irf_loc[main_irf_nr]
```

Comment:
Checking for a shape attribute is not sufficient, a xarray dataarray or numpy array may be 0 dimensions (just a scalar) which cannot be indexed resulting in an error `IndexError("too many indices") `

Reply:
Thanks, nice catch didn't know that the `numpy` scalars also had a `shape` property.